### ISSUE TITLE
[grammar] .Hash method chain with newline

### ISSUE THREAD

**Issue by habere-et-dispertire** [rakudo/rakudo#5369]
A method chain of :

~~~~raku
<Hello, World!>.antipairs.Hash{ 'World!' };
# 1
~~~~

Fails when split across a newline :

~~~~raku
<Hello, World!>.antipairs
               .Hash{ 'World!' };
~~~~

## Error

~~~text
===SORRY!=== Error while compiling
Unexpected block in infix position (missing statement control word before the expression?)
------>                .Hash<HERE>{ 'World!' };
    expecting any of:
        infix
        infix stopper
~~~~

**Comment by jubilatious1** [comment]
Not even a newline; a simple space after `antipairs` throws an error (tested with "unspace"):
```
(<Hello, World!>.antipairs\ .Hash{ 'World!' }).say;  # returns 1
```

But: 

```
(<Hello, World!>.antipairs .Hash{ 'World!' }).say;  # returns:
Unexpected block in infix position (missing statement control word before the expression?)
at -:1
------> [32m(<Hello, World!>.antipairs .Hash[33m⏏[31m{ 'World!' }).say;[0m
    expecting any of:
        infix
        infix stopper
```

**Comment by 2colours** [comment]
There is no need for `<Hello, World!>.antipairs`, you could use any literal, followed by any detached method.

Actually, are there _any_ post(circum)fix operators that work after a detached method call, besides other method calls? You can think of `.[..]`, `.<...>` with or without the dot, you can think of postfix `++`, anything is good to go. Is here a single one that works after a detached method call?

**Comment by jubilatious1** [comment]
I have to agree with the OP that there's something different about the `Hash` call...it seems intolerant to any leading whitespace:

In the REPL (2023.05):
```raku
 > my $a = 'cat'
cat
 > $a .antipairs
(cat => 0)
 > $a .antipairs
(cat => 0)
 > $a .antipairs .Hash{ 'cat' }
===SORRY!=== Error while compiling:
Unexpected block in infix position (missing statement control word before the expression?)
------> $a .antipairs .Hash⏏{ 'cat' }
    expecting any of:
        infix
        infix stopper
 > $a .antipairs.Hash{ 'cat' }
===SORRY!=== Error while compiling:
Unexpected block in infix position (missing statement control word before the expression?)
------> $a .antipairs.Hash⏏{ 'cat' }
    expecting any of:
        infix
        infix stopper
 > $a.antipairs.Hash{ 'cat' }
0
 >
```

**Comment by 2colours** [comment]
It's not the `Hash` call, it's the post(circum)fix operator after a detached method call. The error messages may vary but ultimately, the point is that there seems to be no rule that could make any sense of {} as indexing. Or <> as indexing, or [] as indexing, or ++ for postfix increment etc, while the variants that have a dot in front of them are explicitly forbidden for some reason, even though they could match

This part of the grammar is not easy to read but the bigger problem is, it's full of ad-hoc paths and gibberish terms (dottyopish, fiddly, nibble, babble, etc.)

Originally, detaching method calls wasn't allowed at all, and this probably has grammar implications that persist. As far as I can tell, it's the `postfixish` (lol) rule that deliberately fails to match if there is a whitespace under the cursor, meaning that the parsing falls back to the exotic path of the unsung "infix dot". This will outright eliminate all expressions that don't continue with an alphabetic character (after some whitespaces), and it basically doesn't allow chaining. I don't completely get how but from this point on, it seems to me that from this point on, the "postfixish" matching won't return, it's like you let go of defining a term and now you have to move on with an operator.

**Comment by jubilatious1** [comment]
#2092

**Comment by 2colours** [comment]
Yep, also found it... also, it's on my [issue list](https://github.com/2colours/Raku-ideas/blob/main/Issues.md) for a long time. I wonder why they didn't conclude that maybe the behavior is the problem, and not the message. Also, regarding the message, I wouldn't think that detachment is a transitive property, so in `foo .bar.&baz`, I don't perceive `.&baz` as "detached"; it couldn't be more attached, in fact.

**Comment by jubilatious1** [comment]
Maybe suggest the user can work around the error by "unspacing" before a non-alpha method call?
```raku
[1] > 'kitty' .&uc
===SORRY!=== Error while compiling:
Malformed postfix call (only basic method calls that exclusively use a dot can be detached)
------> 'kitty' .⏏&uc
[1] > 'kitty'\ .&uc
KITTY
[2] >
```

**Comment by habere-et-dispertire** [comment]
I [stumbed upon](https://exercism.org/tracks/raku/exercises/resistor-color-duo/solutions/habere-et-dispertire) this with :

~~~~raku
# https://exercism.org/tracks/raku/exercises/resistor-color-duo/solutions/habere-et-dispertire
#
sub decoded-value ( *@colors ) is export {

    < black brown red orange yellow green blue violet grey white >
        .antipairs andthen
        .Hash{ @colors[ ^ 2 ] }
        .join
        .Int

}
~~~~

I found the use of  `andthen` a workaround that preserves the method chain layout.

**Comment by habere-et-dispertire** [comment]
Seems to work in a recent `rakupp` ! 🪅

~~~~bash
Raku++ 4.0.1 — \h for help, ^D to exit
> say <Hello, World!> .antipairs .Hash{ 'World!' }
1
~~~~
